src/garage/tf/models/categorical_gru_model.py

```diff
@@ -0,0 +1,120 @@
+            initial hidden state. The functino should return a tf.Tensor.
+        hidden_state_init_trainable (bool): Bool for whether the initial
+            hidden state is trainable.
+        layer_normalization (bool): Bool for using layer normalization or not.
+
+    """
+
+    def __init__(self,
+                 output_dim,
+                 hidden_dim,
+                 name=None,
+                 hidden_nonlinearity=tf.nn.tanh,
+                 hidden_w_init=tf.initializers.glorot_uniform(),
+                 hidden_b_init=tf.zeros_initializer(),
+                 recurrent_nonlinearity=tf.nn.sigmoid,
+                 recurrent_w_init=tf.initializers.glorot_uniform(),
+                 output_nonlinearity=None,
+                 output_w_init=tf.initializers.glorot_uniform(),
+                 output_b_init=tf.zeros_initializer(),
+                 hidden_state_init=tf.zeros_initializer(),
+                 hidden_state_init_trainable=False,
+                 layer_normalization=False):
+        super().__init__(
+            output_dim=output_dim,
+            hidden_dim=hidden_dim,
+            name=name,
+            hidden_nonlinearity=hidden_nonlinearity,
+            hidden_w_init=hidden_w_init,
+            hidden_b_init=hidden_b_init,
+            recurrent_nonlinearity=recurrent_nonlinearity,
+            recurrent_w_init=recurrent_w_init,
+            output_nonlinearity=output_nonlinearity,
+            output_w_init=output_w_init,
+            output_b_init=output_b_init,
+            hidden_state_init=hidden_state_init,
+            hidden_state_init_trainable=hidden_state_init_trainable,
+            layer_normalization=layer_normalization)
+
+    def network_output_spec(self):
+        """Network output spec.
+
+        Returns:
+            list[str]: Name of the model outputs, in order.
+
+        """
+        return ['dist', 'step_output', 'step_hidden', 'init_hidden']
+
+    # pylint: disable=arguments-differ
+    def _build(self, state_input, step_input, step_hidden, name=None):
+        """Build model.
+
+        Args:
+            state_input (tf.Tensor): Full observation input, with shape
+                :math: `(N, T, S^*)`.
+            step_input (tf.Tensor): Step observation input, with shape
+                :math: `(N, S^*)`.
+            step_hidden (tf.Tensor): Hidden state for step, with shape
+                :math: `(N, S^*)`.
+            name (str): Inner model name, also the variable scope of the
+                inner model, if exist. One example is
+                garage.tf.models.Sequential.
+
+        Returns:
+            tfp.distributions.Categorical: Policy distribution.
+            tf.Tensor: Step output, with shape :math: `(N, S^*)`.
+            tf.Tensor: Step hidden state, with shape :math: `(N, S^*)`.
+            tf.Tensor: Initial hidden state , used to reset the hidden state
+                when policy resets. Shape: :math: `(S^*)`.
```

Comment:
do these render correctly? i think `:math:` needs to have no space before the `, but not positive. you can test with `make docs`

Reply:
Yes they do need to have no space. Will fix.
Another issue is private functions will not be rendered.